src/Optimizely/Entity/Variation.php

```diff
@@ -95,12 +95,19 @@ public function getVariables(){
         return $this->_variableUsageInstances;
     }
 
+    public function getVariableUsageById($variableId){
+        if(isset($this->_variableIdToVariableUsageInstanceMap[$variableId]))
+            return $this->_variableIdToVariableUsageInstanceMap[$variableId];
+        else
+            return null;
+    }
+
     public function setVariables($variableUsageInstances){
         $this->_variableUsageInstances = ConfigParser::generateMap($variableUsageInstances,null,VariableUsage::class);
 
         if(!empty($this->_variableUsageInstances)){
             foreach(array_values($this->_variableUsageInstances) as $variableUsage){
-                $_variableIdToVariableUsageInstanceMap[$variableUsage->getId()] = $variableUsage;
+                $this->_variableIdToVariableUsageInstanceMap[$variableUsage->getId()] = $variableUsage;
```

Comment:
line 55 to 58, code is repeated, i would say to create a method called generatevariableid tovariableusageinstance and call this method on both places.

Reply:
Done